pvlib/irradiance.py

```diff
@@ -1194,6 +1194,35 @@ def perez(surface_tilt, surface_azimuth, dhi, dni, dni_extra,
         return sky_diffuse
 
 
+def clearsky_index(ghi, clearsky_ghi, max_clearsky_index=2.0):
+    """
+    Calculate the clearsky index.
+
+    The clearsky index is the ratio of global to clearsky global irradiance.
+
+    Parameters
+    ----------
+    ghi : numeric
+        Global horizontal irradiance in W/m^2.
+
+    clearsky_ghi : numeric
+        Modeled clearsky GHI
+
+    max_clearsky_index : numeric, default 2.0
+        Maximum value of the clearsky index. The default, 2.0, allows
+        for over-irradiance events typically seen in sub-hourly data.
+
+    Returns
+    -------
+    clearsky_index : numeric
+        Clearsky index
+    """
+    clearsky_index = ghi * 1. / clearsky_ghi
```

Comment:
How should we handle:

* 0 / 0
* 1 / 0
* -1 / 0

I suggest all return 0.

Reply:
why `* 1.`? If this is important for something (types? broadcasting?) we need a comment to explain why, otherwise someone may change it when trying to clean up code in the future.

Reply:
I followed the lead of `irradiance.clearness_index`, which will return NaN for some inputs.  I'd be happy to add NaN correction here if you think it's appropriate.  

For the `* 1.`, I was worried about the special case of Python `2.x` and the user passing in two integers instead of an np array/pd series.  Is this corner case worth it?  Let me know if I should remove it or document it.

Reply:
If I remember correctly, `clearness_index` returns 0 for night time values and should only return `nan` when `solar_zenith=90` and `min_cos_zenith` is changed from the default. 

We use `from __future__ import division` at the top of the module to account for the possibility of integer division, so I think the `*1.` is not necessary.

Reply:
I updated the behavior to set nan/inf to zero as you suggested, and removed the float conversion -- I had forgotten about the division import.